src/redux/books/booksSlice.js

```diff
@@ -0,0 +1,17 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = [];
+
+export const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.push(action.payload);
+    },
+    removeBook: (state, action) => state.filter((book) => book.id !== action.payload),
+  },
```

Comment:
- [OPTIONAL] As I pointed out above about updating your initial state, you need to update your reducer actions to meet up with that. Instead of pushing directly into the state, you then need to push into the books array in the state object. :+1: :smile: 
```javascript
    addBook: (state, action) => {
      state.books.push(action.payload);
    },
    removeBook: (state, action) => state.books.filter((book) => book.id !== action.payload),
  },

```